frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.unit.spec.tsx

```diff
@@ -0,0 +1,156 @@
+      },
+    });
+    await userEvent.type(input, "foo", {
+      pointerEventsCheck: 0,
+    });
+    input.blur();
+
+    expect(onChange).toHaveBeenCalledWith(["foo"]);
+    expect(input).toHaveValue("");
+
+    // this one does _not_ trigger a change
+    await userEvent.type(input, "bar", {
+      pointerEventsCheck: 0,
+    });
+    input.blur();
+
+    expect(onChange).toHaveBeenCalledWith(["foo"]);
+    expect(input).toHaveValue("");
+  });
+
+  it("should accept values when entering a comma", async () => {
+    const { input, onChange } = setup({ data: EXAMPLE_DATA });
+    await userEvent.type(input, "foo,", {
+      pointerEventsCheck: 0,
+    });
+    expect(onChange).toHaveBeenCalledWith(["foo"]);
+    expect(input).toHaveValue("");
+
+    await userEvent.type(input, "bar,", {
+      pointerEventsCheck: 0,
+    });
+    expect(onChange).toHaveBeenCalledWith(["foo", "bar"]);
+    expect(input).toHaveValue("");
+  });
+
+  it("should accept values with spaces in them", async () => {
+    const { input, onChange } = setup({ data: EXAMPLE_DATA });
+    await userEvent.type(input, "foo bar,", {
+      pointerEventsCheck: 0,
+    });
+    expect(onChange).toHaveBeenCalledWith(["foo bar"]);
+    expect(input).toHaveValue("");
+  });
+
+  it("should not accept values when entering a comma if they are not accepted by shouldCreate", async () => {
+    const { input, onChange } = setup({
+      data: EXAMPLE_DATA,
+      shouldCreate(value: string) {
+        return value === "foo";
+      },
+    });
+    await userEvent.type(input, "foo,", {
+      pointerEventsCheck: 0,
+    });
+    expect(onChange).toHaveBeenCalledWith(["foo"]);
+    expect(input).toHaveValue("");
+
+    // this one does _not_ trigger a change
+    await userEvent.type(input, "bar,", {
+      pointerEventsCheck: 0,
+    });
+    expect(onChange).toHaveBeenLastCalledWith(["foo"]);
+    expect(input).toHaveValue("");
+  });
+
+  it("should accept comma-separated values when pasting", async () => {
+    const { input, onChange } = setup({ data: EXAMPLE_DATA });
+    input.focus();
+    await userEvent.paste("foo,bar");
+    expect(onChange).toHaveBeenCalledWith(["foo", "bar"]);
+    expect(input).toHaveValue("");
+  });
+
+  it("should accept newline-separated values when pasting", async () => {
+    const { input, onChange } = setup({ data: EXAMPLE_DATA });
+    input.focus();
+    await userEvent.paste("foo\nbar");
+    expect(onChange).toHaveBeenCalledWith(["foo", "bar"]);
+    expect(input).toHaveValue("");
+  });
+
+  it("should accept comma-separated values, but omit values not accepted by shouldCreate", async () => {
+    const { input, onChange } = setup({
+      data: EXAMPLE_DATA,
+      shouldCreate(value: string) {
+        return value === "foo" || value === "bar";
+      },
+    });
+    input.focus();
+    await userEvent.paste("foo,bar,baz");
+    expect(onChange).toHaveBeenCalledWith(["foo", "bar"]);
+    expect(input).toHaveValue("");
+  });
+});
```

Comment:
nit:
1. copy this: `xyz, 123`
2. type `abc` into the input
3. paste


Expected result (either one will do):
- 2 values selected (`abcxyz`, `123`)
- 3 values selected (`abc`, `xyz`, `123`)
- 2 values selected (`xyz`, `123`) and `abc` still unsubmitted


Actual result: 3 values selected (`abc`, `xyz`, `123`) and `abc` still as 4th unsubmitted value

Reply:
This is fixed